lib/internal/mime.js

```diff
@@ -0,0 +1,318 @@
+  target.subtype = subtype;
+  target.parametersString = StringPrototypeSlice(str, position);
+  return target;
+}
+
+function parseParametersString(str) {
+  const paramsMap = new Map();
+  str = StringPrototypeReplace(str, ENDING_WHITESPACE, '');
+  let position = 0;
+  while (position < str.length) {
+    // Skip any whitespace before parameter
+    const skip = execPattern(
+      BEGINNING_WHITESPACE,
+      StringPrototypeSlice(str, position))[0];
+    if (skip) position += skip.length;
+    // Read until ';' or '='
+    const parameterMatch = execPattern(
+      /^[^;=]*/,
+      StringPrototypeSlice(str, position));
+    const parameterString = toASCIILower(parameterMatch[0]);
+    position += parameterString.length;
+    // If we found a terminating character
+    if (position < str.length) {
+      // Safe to use because we never do special actions for surrogate pairs
+      const char = StringPrototypeCharAt(str, position);
+      // Skip the terminating character
+      position += 1;
+      // Ignore parameters without values
+      if (char === ';') {
+        continue;
+      }
+    }
+    // If we are at end of the string, it cannot have a value
+    if (position >= str.length) break;
+    // Safe to use because we never do special actions for surrogate pairs
+    const char = StringPrototypeCharAt(str, position);
+    let parameterValue = null;
+    if (char === '"') {
+      // Handle quoted-string form of values
+      // skip '"'
+      position += 1;
+      // Find matching closing '"' or end of string
+      //   use $1 to see if we terminated on unmatched '\'
+      //   use $2 to see if we terminated on a matching '"'
+      //   so we can skip the last char in either case
+      const inside = execPattern(
+        /^(?:([\\]$)|[\\][\s\S]|[^"])*(?:(")|$)/u,
+        StringPrototypeSlice(str, position));
+      position += inside[0].length;
+      // Unescape '\' quoted characters
+      parameterValue = StringPrototypeReplace(StringPrototypeSlice(
+        inside[0],
+        0,
+        // Skip including last character if an unmatched '\' or '"' during
+        // unescape
+        inside[1] || inside[2] ? -1 : undefined
+      ), /[\\]([\s\S])/ug, '$1');
+      // If we did have an unmatched '\' add it back to the end
+      if (inside[1]) parameterValue += '\\';
+    } else {
+      // Handle the normal parameter value form
+      let value = execPattern(/[^;]*/, StringPrototypeSlice(str, position))[0];
+      position += value.length;
+      value = StringPrototypeReplace(value, ENDING_WHITESPACE, '');
+      // Ignore parameters without values
+      if (value === '') continue;
+      parameterValue = value;
+    }
+    if (parameterString !== '' &&
+      testPattern(NotHTTPTokenCodePoint, parameterString) === false &&
+      testPattern(NotHTTPQuotedStringCodePoint, parameterValue) === false &&
+      paramsMap.has(parameterString) === false) {
+      paramsMap.set(parameterString, parameterValue);
```

Comment:
Also, not an expert with MIME but is it okay that the parameter will be just ignored (without error) if it is already present in the map?

Reply:
this is the expected behavior per https://mimesniff.spec.whatwg.org/

Reply:
i might have misunderstood, are you asking why params are only ever 1-1 mappings, and if we replace a mapping why it doesn't error? that just matches things like `URLSearchParams` and `Map`.

Reply:
No, the first answer is what I wanted, thanks. Just wanted to make sure.